### PR TITLE
feat(amplify-provider-awscloudformation): assume role

### PR DESCRIPTION
**Pull request by UnleashedMind**
*Issue #, if available:*
#613 
*Description of changes:*
allows cli to assume IAM role to access aws resources, as described in [this doc](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-role.html#cli-configure-role-mfa)

Note that this implementation does not include using MFA

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/lib/configuration-manager.js [hunk 1]
```diff
@@ -357,44 +357,39 @@ function removeProjectConfig(context) {
...
-  const { config } = projectConfigInfo;
-  if (config.useProfile) {
-    return systemConfigManager.getProfile(config.profileName).region;
+  const profileName = process.env.AWS_PROFILE || 'default';
+  const profileRegion = systemConfigManager.getProfileRegion(profileName);
+  if (profileRegion) {
+    return profileRegion;
   }
-  return config.region;
+  return 'us-east-1';
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/lib/configuration-manager.js]: Why should it always return us-east-1?

File: packages/amplify-provider-awscloudformation/lib/configuration-manager.js [hunk 2]
```diff
@@ -357,44 +357,39 @@ function removeProjectConfig(context) {
...
-  }
+
   return awsClient;
 }
 
-function getRegion(projectConfigInfo) {
+function getRegion() {
   // For details of how aws region is set, check the following link
   // https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/setting-region.html
-  if (projectConfigInfo.configLevel === 'general') {
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/lib/configuration-manager.js]: Why is this if statement remove?

**Comment by kaustavghosh06** [comment]
Looking at the PR right now, could you also make the corresponding documentation changes for this change?

File: packages/amplify-provider-awscloudformation/lib/system-config-manager.js [hunk 3]
```diff
@@ -54,76 +55,107 @@ function setProfile(awsConfig, profileName) {
...
-
-    awsConfigs = {};
 
     Object.keys(credentials).forEach((key) => {
-      const profileName = key.trim();
-      awsConfigs[profileName] = {
-        accessKeyId: credentials[key].aws_access_key_id,
-        secretAccessKey: credentials[key].aws_secret_access_key,
-      };
+      const keyName = key.trim();
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/lib/system-config-manager.js]: This variable could be more descriptive.

File: packages/amplify-provider-awscloudformation/lib/system-config-manager.js [hunk 4]
```diff
@@ -54,76 +55,107 @@ function setProfile(awsConfig, profileName) {
...
+    if (!isRoleSourceProfile && profileConfig.role_arn && profileConfig.source_profile) {
+      const roleCredentials =
+        await getRoleCredentials(profileConfig);
+      delete profileConfig.role_arn;
+      delete profileConfig.source_profile;
+      awsConfig = {
+        ...profileConfig,
+        ...roleCredentials,
+      };
+    } else {
```
Comment (kaustavghosh06) [inline packages/amplify-provider-awscloudformation/lib/system-config-manager.js]: Not sure as to in which case this branch would get hit?

File: packages/amplify-provider-awscloudformation/lib/configuration-manager.js (hunk 1 above)
Comment (kaustavghosh06) [reply packages/amplify-provider-awscloudformation/lib/configuration-manager.js]: If it's 'general', shouldn't the region be present in either the env variables or the default?

File: packages/amplify-provider-awscloudformation/lib/system-config-manager.js (hunk 4 above)
Comment (UnleashedMind) [reply packages/amplify-provider-awscloudformation/lib/system-config-manager.js]: it is hit for regular profile that does not use roles.

File: packages/amplify-provider-awscloudformation/lib/configuration-manager.js (hunk 1 above)
Comment (UnleashedMind) [reply packages/amplify-provider-awscloudformation/lib/configuration-manager.js]: removed default and throw error if region can not be resolved.

Comment (UnleashedMind) [reply packages/amplify-provider-awscloudformation/lib/configuration-manager.js]: I refactored this part. 
I removed attach region flag for the method, and let the caller (aws-pinpoint.js) handle the specifics, this is now much cleaner.

**Pull request merged**